### PR TITLE
Scripts to demonstrate how to send logs to the logs socket intake

### PR DESCRIPTION
**Pull request by davidjudge1965**
Created a couple scripts in Python and node.js to demonstrate to a customer how they can send logs to the intake socket.

### CODE REVIEW

File: logs_intake/socket/SendLogEntriesOverSocket.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+}
+
+host = "intake.logs.datadoghq.com"
+ssl_port = 10516
+
+DD_SOURCE = "ddsource"
+DD_CUSTOM_TAGS = "ddtags"
+
+metadata[DD_SOURCE] = "apigee"
+metadata[DD_CUSTOM_TAGS] = "level:info,x.api:identity-v1,env:dev,project:apigee"
```
Comment (estib) [inline logs_intake/socket/SendLogEntriesOverSocket.py]: ditto from above comments, maybe like this?

```
metadata["ddtags"] = "level:info,x.api:identity-v1,env:dev,project:apigee"
```

File: logs_intake/socket/SendLogEntriesOverSocket.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+
+def lambda_handler(event):
+    # Check prerequisites
+    if ddApiKey == "<your_api_key>" or ddApiKey == "":
```
Comment (estib) [inline logs_intake/socket/SendLogEntriesOverSocket.py]: the user has already defined `ddApiKey` up top, why are we having them define it again here?

File: logs_intake/socket/SendLogEntriesOverSocket.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+ssl_port = 10516
+
+DD_SOURCE = "ddsource"
+DD_CUSTOM_TAGS = "ddtags"
+
+metadata[DD_SOURCE] = "apigee"
+metadata[DD_CUSTOM_TAGS] = "level:info,x.api:identity-v1,env:dev,project:apigee"
+
+
+def lambda_handler(event):
```
Comment (estib) [inline logs_intake/socket/SendLogEntriesOverSocket.py]: why are we calling this `lambda_handler`?

File: logs_intake/socket/SendLogEntriesOverSocket.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+                raise Exception(
+                    'Conflict while merging metadatas and the log entry at %s' % '.'.join(path + [str(key)])
+                )
+        else:
+            a[key] = b[key]
+    return a
+
+if __name__ == "__main__":
+    # execute only if run as a script
+    metadata[DD_CUSTOM_TAGS] = "x.error,level:error,x.api-identity-v1,x.api:identity-v1,x.env-prd,x.env:prd,x.proj-apigee,x.proj:apigee"
```
Comment (estib) [inline logs_intake/socket/SendLogEntriesOverSocket.py]: what are all these? are these supposed to be examples of the script in action? can we please dumb it down to make it easier for the reader to modify to fit their needs?

File: logs_intake/socket/SendLogEntriesOverSocket.py [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    lambda_handler({"apiProduct":"product-v1-prod","appName":"morrisons-xxwmm-basketgateway_service-prod","clientIp":"52.49.111.54","environment":"prd","clientRequestResource":"/items/314","clientRequestSize":0,"clientRequestUri":"/product/v1/items/314","clientRequestQuerystring":"fields=itemDescription%2CcommercialHierarchy&apikey=m6jR90jCJtASGUArO4vik5BpsFTVIEj6","clientRequestVerb":"GET","clientRequestTime":1524836354312,"responseCode":404,"backendResponseCode":404,"timeBackend":5,"timeTotal":10,"errorMessage":"Item not found","errorCode":"404.19.001","userAgent":"AmazonSQSBufferedAsyncClient/1.11.123"})
```
Comment (estib) [inline logs_intake/socket/SendLogEntriesOverSocket.py]: it appears as though there are some potentially sensitive strings in these example arguments? like api key fields? we should probably scrub this... in fact, this is a public repo, so you may have already leaked something even by pushing this branch...

this and the last comment apply to the next few lines as well.

File: logs_intake/socket/writeToTLSSocket.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
+tls = require('tls');
+var host = "intake.logs.datadoghq.com"
+var ssl_port = 10516
+
+// callback for when secure connection established
+function connected(stream) {
```
Comment (estib) [inline logs_intake/socket/writeToTLSSocket.js]: this function is beautifully simple. can we not make the python example as beautifully simple?

File: logs_intake/socket/writeToTLSSocket.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    if (stream) {
+      // socket connected
+      // REPLACE THE API KEY with your own:
+      api_key= "<your_api_key>"
+      stream.write(api_key + " " + '{"ddsourcecategory": "aws", "appName": "morrisons-xxwmm-storefinder-prod", "clientRequestResource": "/stores/373/specialopeningtimes", "timeBackend": 6, "clientRequestQuerystring": "apikey=kxBdM2chFwZjNvG2PwnSn3sj6C53dLEY", "backendResponseCode": 200, "clientRequestTime": 1524836411874, "clientRequestVerb": "GET", "clientRequestSize": 0, "apiProduct": "location-v2-prod", "ddsource": "apigee", "ddtags": "x.info,level:info,x.api-identity-v1,x.api:identity-v1,x-env-dev,x-env:dev,x.proj-apigee,x.proj:apigee", "clientRequestUri": "/location/v2/stores/373/specialopeningtimes", "environment": "prd", "responseCode": 200, "clientIp": "91.208.239.126", "timeTotal": 9}'+"\n");
```
Comment (estib) [inline logs_intake/socket/writeToTLSSocket.js]: there appears to be some sensitive strings in this, apiKey or whatever. can we scrub that?

Also, this is a very specific log event being sent. for miscelany scripts, we want to make it easy for people to understand what each part of the content is, so we should probably fill it with dummy info. Like "message": "put your log line here", "ddsourcecategory": "optionally add a source category here", etc.

breaking each attribute onto its own line would probably help make it clearer too.

File: logs_intake/socket/SendLogEntriesOverSocket.py [hunk 8]
```diff
@@ -0,0 +1,104 @@
+# This code is provided as an example only.
+import json
+import time
+import socket
+import ssl
+
+ddApiKey = "<your_api_key>"
+
+# metadata: Additional metadata to send with the logs
```
[the remark this answers is not in the document]
Comment (davidjudge1965) [reply logs_intake/socket/SendLogEntriesOverSocket.py]: Done

File: logs_intake/socket/SendLogEntriesOverSocket.py [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+metadata = {
+    "ddsourcecategory": "aws",
+}
+
+host = "intake.logs.datadoghq.com"
+ssl_port = 10516
+
+DD_SOURCE = "ddsource"
```
[the remark this answers is not in the document]
Comment (davidjudge1965) [reply logs_intake/socket/SendLogEntriesOverSocket.py]: Done

File: logs_intake/socket/SendLogEntriesOverSocket.py (hunk 3 above)
Comment (davidjudge1965) [reply logs_intake/socket/SendLogEntriesOverSocket.py]: Because that's what it was called when I borrowed the code - it came from our AWS logs lambda function.

File: logs_intake/socket/SendLogEntriesOverSocket.py (hunk 2 above)
Comment (davidjudge1965) [reply logs_intake/socket/SendLogEntriesOverSocket.py]: It's a test, not a definition.  As this code may be used by someone else, we need to check that the value has been provided - otherwise when they run the code and get success but no data appearing in DD, it's confusing.  This way if they haven't set the key a few lines earlier, they get a useful error message.

File: logs_intake/socket/SendLogEntriesOverSocket.py (hunk 4 above)
Comment (davidjudge1965) [reply logs_intake/socket/SendLogEntriesOverSocket.py]: Cleaned-up somewhat.  Reduced down to 2 examples.

File: logs_intake/socket/SendLogEntriesOverSocket.py (hunk 5 above)
Comment (davidjudge1965) [reply logs_intake/socket/SendLogEntriesOverSocket.py]: API keys and customer name removed.  I don't know how to check if there is any history that also needs removing.

File: logs_intake/socket/writeToTLSSocket.js (hunk 6 above)
Comment (davidjudge1965) [reply logs_intake/socket/writeToTLSSocket.js]: The python code shows how to merge tags into the logs being sent and does some error checking.

File: logs_intake/socket/SendLogEntriesOverSocket.py (hunk 4 above)
Comment (davidjudge1965) [reply logs_intake/socket/SendLogEntriesOverSocket.py]: Made examples simpler

**Comment by davidjudge1965** [comment]
Closing the PR so I can delete the branch

**Comment by davidjudge1965** [comment]
Not needed right now

**Comment by davidjudge1965** [comment]
Cleaning up

**Comment by davidjudge1965** [comment]
Still looking for the solution
